File: tests/benchmarks/resolution/fixtures/kotlin/Repository.kt
package benchmark

data class User(val name: String, val email: String)

open class Repository {
    private val store = mutableListOf<User>()

    open fun save(user: User): Boolean {
        store.add(user)
        return true
    }

    open fun findByName(name: String): User? {
        return store.firstOrNull { it.name == name }
    }

    open fun delete(name: String): Boolean {
        return store.removeIf { it.name == name }
    }
}

class UserRepository : Repository() {
    fun saveIfValid(user: User): Boolean {
        if (!validateName(user.name)) return false
        if (!validateEmail(user.email)) return false
        return save(user)
    }
}

File: tests/benchmarks/resolution/fixtures/csharp/Repository.cs
using System.Collections.Generic;

namespace Benchmark;

public interface IRepository
{
    User FindById(string id);
    void Save(User user);
    bool Delete(string id);
}

public class UserRepository : IRepository
{
    private readonly Dictionary<string, User> _store = new();

    public User FindById(string id)
    {
        _store.TryGetValue(id, out var user);
        return user;
    }

    public void Save(User user)
    {
        _store[user.Id] = user;
    }

    public bool Delete(string id)
    {
        return _store.Remove(id);
    }
}

public class User
{
    public string Id { get; set; }
    public string Name { get; set; }
    public string Email { get; set; }
}
